src/Meilisearch/MeilisearchClient.cs

```diff
@@ -122,34 +122,27 @@ public async Task<TaskInfo> DeleteIndexAsync(string uid, CancellationToken cance
         /// </summary>
         /// <param name="cancellationToken">The cancellation token for this call.</param>
         /// <returns>An IEnumerable of indexes in JsonElement format.</returns>
-        public async Task<IEnumerable<JsonElement>> GetAllRawIndexesAsync(CancellationToken cancellationToken = default)
+        public async Task<JsonDocument> GetAllRawIndexesAsync(CancellationToken cancellationToken = default)
         {
             var response = await _http.GetAsync("indexes", cancellationToken).ConfigureAwait(false);
 
             var content = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
-            var json = JsonDocument.Parse(content);
-            var indexes = new List<JsonElement>();
-
-            foreach (var element in json.RootElement.EnumerateArray())
-            {
-                indexes.Add(element);
-            }
-
-            return indexes;
+            return JsonDocument.Parse(content);
         }
 
         /// <summary>
         /// Gets all the Indexes for the instance. Throws error if the index does not exist.
         /// </summary>
         /// <param name="cancellationToken">The cancellation token for this call.</param>
         /// <returns>Return Enumerable of Index.</returns>
-        public async Task<IEnumerable<Index>> GetAllIndexesAsync(CancellationToken cancellationToken = default)
+        public async Task<ResourceResults<IEnumerable<Index>>> GetAllIndexesAsync(CancellationToken cancellationToken = default)
         {
             var response = await _http.GetAsync("indexes", cancellationToken).ConfigureAwait(false);
 
-            var content = await response.Content.ReadFromJsonAsync<IEnumerable<Index>>(cancellationToken: cancellationToken).ConfigureAwait(false);
-            return content
+            var content = await response.Content.ReadFromJsonAsync<ResourceResults<IEnumerable<Index>>>(cancellationToken: cancellationToken).ConfigureAwait(false);
+            content.Results
```

Comment:
I'm not so sure about what this line 143 is doing... Can you check if it is important?

Because I assume the result from the LINQ query `.Select` will not be applied to the `content` variable.

Reply:
Yes, I had an issue where it should add the `HttpClient _http` but I'm not sure it works well ...

Reply:
I think I got it, it adds the HTTP client as a field/attribute to the instance of that object, in this case, the index and the user can make requests normally without worrying about the `HttpClient`.

Reply:
It should but It doesn't seem to work properly inside the code even if I try the function outside and it works fine